### PR TITLE
SageMaker Training Sensors and tests

### PR DESCRIPTION
**Pull request by troychen728**
Make sure you have checked _all_ steps below.

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-XXX
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/sensors/sagemaker_base_sensor.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# specific language governing permissions and limitations
+# under the License.
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils import apply_defaults
+from airflow.exceptions import AirflowException
+
+
+class SageMakerBaseSensor(BaseSensorOperator):
+    """
+    Contains general sensor behavior for EMR.
```
Comment (yangaws) [inline airflow/contrib/sensors/sagemaker_base_sensor.py]: EMR？

File: airflow/contrib/sensors/sagemaker_base_sensor.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            *args, **kwargs):
+        super(SageMakerBaseSensor, self).__init__(*args, **kwargs)
+        self.aws_conn_id = aws_conn_id
+
+    def poke(self, context):
+        try:
+            response = self.get_sagemaker_response()
+        except AttributeError:
+            raise AirflowException(
+                "Sagemaker sensor subclass get response function not implemented.")
```
Comment (yangaws) [inline airflow/contrib/sensors/sagemaker_base_sensor.py]: "Method get_sagemaker_response() not implemented" is good enough

File: airflow/contrib/sensors/sagemaker_base_sensor.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        if not response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            self.log.info('Bad HTTP response: %s', response)
+            return False
+        try:
+            state = self.state_from_response(response)
+        except ValueError:
+            raise AirflowException(
+                "Sagemaker sensor subclass get state function not implemented.")
```
Comment (yangaws) [inline airflow/contrib/sensors/sagemaker_base_sensor.py]: Similar here

File: tests/contrib/sensors/test_sagemaker_base_sensor.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+        operator = SageMakerBaseSensorSubclass(
+            task_id='test_task',
+            poke_interval=2,
+            aws_conn_id='aws_test'
+        )
+
+        self.assertEqual(operator.poke(None), False)
+
+    def test_poke_returns_false_when_method_not_implemented(self):
```
Comment (yangaws) [inline tests/contrib/sensors/test_sagemaker_base_sensor.py]: I think it doesn't return false. Should be test_poke_raise_exception_......

File: tests/contrib/sensors/test_sagemaker_base_sensor.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            def get_sagemaker_response(self):
+                return {
+                    'SomeKey': {'State': 'COMPLETED'},
+                    'ResponseMetadata': {'HTTPStatusCode': 200}
+                }
+
+            def state_from_response(self, response):
+                return response['SomeKey']['State']
+
+        operator = SageMakerBaseSensorSubclass(
```
Comment (yangaws) [inline tests/contrib/sensors/test_sagemaker_base_sensor.py]: I guess sensor would be a better name than operator here (and in all other tests).

**Review by yangaws — COMMENTED** [review]
Mostly name and doc change.

File: airflow/contrib/sensors/sagemaker_base_sensor.py (hunk 1 above)
Comment (troychen728) [reply airflow/contrib/sensors/sagemaker_base_sensor.py]: Fixed

File: airflow/contrib/sensors/sagemaker_base_sensor.py (hunk 2 above)
Comment (troychen728) [reply airflow/contrib/sensors/sagemaker_base_sensor.py]: Changed the error message

File: airflow/contrib/sensors/sagemaker_base_sensor.py (hunk 3 above)
Comment (troychen728) [reply airflow/contrib/sensors/sagemaker_base_sensor.py]: Changed the error message

File: tests/contrib/sensors/test_sagemaker_base_sensor.py (hunk 4 above)
Comment (troychen728) [reply tests/contrib/sensors/test_sagemaker_base_sensor.py]: changed the method name

File: tests/contrib/sensors/test_sagemaker_base_sensor.py (hunk 5 above)
Comment (troychen728) [reply tests/contrib/sensors/test_sagemaker_base_sensor.py]: changed the variable name to sensor

**Review by yangaws — APPROVED** [review]
Cool!

**Pull request merged**